trestle/core/markdown/control_markdown_node.py

```diff
@@ -17,19 +17,46 @@
 from __future__ import annotations
 
 import logging
+import re
 import string
-from typing import List, Optional, Tuple
+from enum import Enum
+from typing import Dict, List, Optional, Tuple
 
 from trestle.common import const
 from trestle.common.err import TrestleError
-from trestle.common.list_utils import none_if_empty
+from trestle.common.list_utils import as_filtered_list, as_list, none_if_empty
+from trestle.common.str_utils import spaces_and_caps_to_snake
 from trestle.core.control_interface import ControlInterface
 from trestle.core.markdown.base_markdown_node import BaseMarkdownNode, BaseSectionContent
+from trestle.core.markdown.markdown_const import CODEBLOCK_DEF, HTML_COMMENT_END_REGEX, HTML_COMMENT_START
 from trestle.oscal import common
 
 logger = logging.getLogger(__name__)
 
 
+class ControlSectionType(Enum):
+    """A type of the section in the markdown.
+
+    Supported types:
+    1 - Control Statement
+    2 - Control Objective
+    3 - Control Guidance
+    4 - ## Part foo
+    5 - Control foo
+    6 - ### foo
+    7 - # Editable content
+    """
+
+    control_statement = 1
+    control_objective = 2
+    control_guidance = 3
+    editable_by_id_part = 4
+    generic_control_part = 5
+    generic_subpart = 6
+    editable_content = 7
+    undefined = 8
```

Comment:
Please use caps for enums.  I think it's fairly standard convention and it is the convention in trestle - except in the auto-generated OSCAL python classes.  And it would be good to add doc strings on Part vs. part - e.g. is `## part foo` the same as `## Part foo`